### PR TITLE
fix(ts): activate strictNullChecks (#2412)

### PR DESCRIPTION
**Pull request by carinelebas**
# Context
Fix last errors of strictNullChecks:
- use null guards
- make new util requireUser on context and use it since user may not always be defined.
- set strictNullChecks to true

test there is no regression

## Related issues

- Related to #2412

# Checklist

## Quality

- [x] I consider this work complete and ready for review
- [ ] I tested all features and edge cases
- [ ] I refactored code where necessary to improve quality
- [ ] I made sure my code meets development guidelines
- [x] I performed a self-review of my code
- [ ] I added the `skip-feature-env` label if a feature environment is not needed for this PR

## Testing

- [ ] I wrote test cases for the relevant use cases (unit, integration or e2e tests)
- [ ] (Bug fixes only) I added tests that reproduce and verify the fix

### What tests have been made

- [x] Unit tests
- [x] Integration tests
- [ ] E2E tests
- [x] Local manual tests

# Additional information
> Any additional context, dependencies, or concerns reviewers should know about.

### CODE REVIEW

**Comment by SamuelHassine** [comment]
🤖 [AI-generated]

Hey @carinelebas! 👋 I'm Sam, your convention guardian. I reviewed PR #2522 and here's what I adjusted:

### 🏷️ Labels
| Action | Label | Reason |
|---|---|---|
| ✅ Added | `filigran team` | A pull request must carry exactly one ownership label (`filigran team` / `community`); you're an org member, so `filigran team` applies. |

Your title `fix(ts): activate strictNullChecks (#2412)` already follows the convention perfectly — valid `fix:` type, lowercase scope, lowercase description, `strictNullChecks` preserved, and the required `(#2412)` issue reference. Nothing to change there. ✨

> 💡 For future PRs, see the [title & label taxonomy](https://github.com/FiligranHQ/xtm-hub/blob/development/.github/LABELS.md).

Thanks for contributing! 🚀

File: apps/backend/src/modules/deployment/deployment.app.ts [hunk 1]
```diff
@@ -74,11 +74,15 @@ export const DeploymentApp = {
...
     input: CreateDeploymentRequestInput
   ): Promise<DeploymentRequest> => {
-    const { user } = requestContext.require();
+    const user = requestContext.requireUser();
     const chosenOrganization = await OrganizationDomain.loadOrganizationBy({
       id: user.selected_organization_id,
     });
 
+    if (!chosenOrganization) {
+      throw new Error(NotFoundErrorCode.OrganizationNotFound);
```
Comment (Kroustille) [inline apps/backend/src/modules/deployment/deployment.app.ts]:
```suggestion
      throw new Error(ErrorCode.OrganizationNotFound);
```

File: apps/backend/src/modules/deployment/deployment.app.ts [hunk 2]
```diff
@@ -698,12 +702,15 @@ export const DeploymentApp = {
...
   loadTrialDeployments: async (input: TrialDeploymentsInput) => {
-    const { user } = requestContext.require();
+    const user = requestContext.requireUser();
     await securityGuard.assertUserIsInOrganization(user, input.organizationId);
 
     const organization = await OrganizationDomain.loadOrganizationBy({
       id: input.organizationId,
     });
+    if (!organization) {
+      throw new Error(NotFoundErrorCode.OrganizationNotFound);
```
Comment (Kroustille) [inline apps/backend/src/modules/deployment/deployment.app.ts]:
```suggestion
      throw new Error(ErrorCode.OrganizationNotFound);
```

File: apps/backend/src/modules/deployment/deployment.app.ts [hunk 3]
```diff
@@ -971,6 +978,9 @@ const sendUpdateDeploymentTelemetryEvent = async (
     const organization = await OrganizationDomain.loadOrganizationBy({
       id: deploymentRequest.organization_requester_id,
     });
+    if (!organization) {
+      throw new Error(NotFoundErrorCode.OrganizationNotFound);
```
Comment (Kroustille) [inline apps/backend/src/modules/deployment/deployment.app.ts]:
```suggestion
      throw new Error(ErrorCode.OrganizationNotFound);
```

File: apps/backend/tsconfig.json [hunk 4]
```diff
@@ -12,7 +12,7 @@
     "resolveJsonModule": true,
     "moduleDetection": "force",
     "strict": true,
-    "strictNullChecks": false,
+    "strictNullChecks": true,
```
Comment (Kroustille) [inline apps/backend/tsconfig.json:15]: 🎉

File: apps/backend/src/thirdparty/hubspot/hubspot.ts [hunk 5]
```diff
@@ -151,10 +153,9 @@ export const hubspotReachOutSalesHook = async ({
           `Deployment request ${deploymentRequest.id} is not a trial deployment request`
         );
       }
-
+      const request = deploymentRequest;
```
Comment (Kroustille) [inline apps/backend/src/thirdparty/hubspot/hubspot.ts:156]: Why this change ? Isn't it possible to just use `deploymentRequest` ?

File: apps/backend/src/modules/document/document-download-endpoint.ts [hunk 6]
```diff
@@ -124,12 +139,15 @@ export const documentDownloadEndpoint = (app) => {
                 id: user.selected_organization_id,
               });
 
+              if (!selectedOrga) {
+                throw new Error(NotFoundErrorCode.OrganizationNotFound);
```
Comment (Kroustille) [inline apps/backend/src/modules/document/document-download-endpoint.ts]:
```suggestion
                throw new Error(ErrorCode.OrganizationNotFound);
```

File: apps/backend/src/modules/document/document-download-endpoint.ts [hunk 7]
```diff
@@ -83,15 +83,23 @@ export const documentDownloadEndpoint = (app) => {
...
+        logApp.info('Downloading file:', { filename });
 
         try {
           const document = await DocumentDomain.loadDocumentBy({
-            id: fromGlobalId(
-              Array.isArray(req.params.filename)
-                ? req.params.filename[0]
-                : req.params.filename
-            ).id as DocumentId,
+            id: fromGlobalId(filename).id as DocumentId,
```
Comment (Kroustille) [inline apps/backend/src/modules/document/document-download-endpoint.ts]:
```suggestion
            id: extractId<DocumentId>(filename),
```

File: apps/backend/src/modules/organization-management/user/user-organization/user-organization.app.ts [hunk 8]
```diff
@@ -27,14 +30,17 @@ export const UserOrganizationApp = {
...
+    const contextUser = requestContext.requireUser();
     const [organizationFromEmail] =
       await OrganizationDomain.loadOrganizationsFromEmail(input.email);
 
     const chosenOrganization = await OrganizationDomain.loadOrganizationBy({
       id: contextUser.selected_organization_id,
     });
 
+    if (!chosenOrganization) {
+      throw new Error(NotFoundErrorCode.OrganizationNotFound);
```
Comment (Kroustille) [inline apps/backend/src/modules/organization-management/user/user-organization/user-organization.app.ts]:
```suggestion
      throw new Error(ErrorCode.OrganizationNotFound);
```

File: apps/backend/src/modules/organization-management/user/user.test.ts [hunk 9]
```diff
@@ -77,6 +78,9 @@ describe('user helpers', async () => {
       const newOrganization = await OrganizationDomain.loadOrganizationBy({
         name: organizationName,
       });
+      if (!newOrganization) {
+        throw new Error(NotFoundErrorCode.OrganizationNotFound);
```
Comment (Kroustille) [inline apps/backend/src/modules/organization-management/user/user.test.ts]:
```suggestion
        throw new Error(ErrorCode.OrganizationNotFound);
```

File: apps/backend/src/modules/subscription/subscription.app.ts [hunk 10]
```diff
@@ -78,7 +78,7 @@ export const subscriptionApp = {
   deleteSubscriptions: async (
     ids: SubscriptionId[]
   ): Promise<Subscription[]> => {
-    return SubscriptionDomain.deleteSubscriptions(ids);
+    return (await SubscriptionDomain.deleteSubscriptions(ids)) ?? [];
```
Comment (Kroustille) [inline apps/backend/src/modules/subscription/subscription.app.ts]: I think it would be better to change the typing of `SubscriptionDomain.deleteSubscriptions` because I don't think it fits with reality

File: apps/backend/src/modules/subscription/subscription.resolver.ts [hunk 11]
```diff
@@ -35,8 +35,15 @@ const resolvers: Resolvers = {
...
     user_service: ({ id }, _) =>
       SubscriptionDomain.getUserService(id as SubscriptionId),
-    organization: ({ organization_id }, _) =>
-      OrganizationDomain.loadOrganizationBy({ id: organization_id }),
+    organization: async ({ organization_id }, _) => {
+      const orga = await OrganizationDomain.loadOrganizationBy({
+        id: organization_id,
+      });
+      if (!orga) {
+        throw new Error(NotFoundErrorCode.OrganizationNotFound);
```
Comment (Kroustille) [inline apps/backend/src/modules/subscription/subscription.resolver.ts]:
```suggestion
        throw new Error(ErrorCode.OrganizationNotFound);
```

File: apps/backend/src/modules/telemetry/telemetry.app.ts [hunk 12]
```diff
@@ -82,23 +83,33 @@ export const telemetryApp = {
...
   }) {
-    const { user } = requestContext.require();
+    const user = requestContext.requireUser();
     const selected_organization_id = user.selected_organization_id;
 
     const selectedOrga = await OrganizationDomain.loadOrganizationBy({
       id: selected_organization_id,
     });
+    if (!selectedOrga) {
+      throw new Error(NotFoundErrorCode.OrganizationNotFound);
```
Comment (Kroustille) [inline apps/backend/src/modules/telemetry/telemetry.app.ts]:
```suggestion
      throw new Error(ErrorCode.OrganizationNotFound);
```

File: apps/backend/src/modules/telemetry/telemetry.app.ts [hunk 13]
```diff
@@ -82,23 +83,33 @@ export const telemetryApp = {
...
+    }
+
     const serviceDefinition = await loadServiceDefinitionByServiceInstance(
       input.service_instance_id
     );
 
+    if (!serviceDefinition) {
+      throw new Error(NotFoundErrorCode.ServiceNotFound);
```
Comment (Kroustille) [inline apps/backend/src/modules/telemetry/telemetry.app.ts]:
```suggestion
      throw new Error(ErrorCode.ServiceNotFound);
```

File: apps/backend/src/modules/telemetry/telemetry.app.ts [hunk 14]
```diff
@@ -82,23 +83,33 @@ export const telemetryApp = {
...
+    }
     const platformServiceInstanceId = extractId<'RegisteredPlatform'>(
       input.platform_service_instance_id
     );
     const platformConfiguration =
       await loadPlatformConfigurationByServiceInstanceId(
         platformServiceInstanceId
       );
+    if (!platformConfiguration) {
+      throw new Error(NotFoundErrorCode.PlatformConfigurationNotFound);
```
Comment (Kroustille) [inline apps/backend/src/modules/telemetry/telemetry.app.ts]:
```suggestion
      throw new Error(ErrorCode.PlatformConfigurationNotFound);
```

File: apps/backend/src/modules/use-case/use-case.app.ts [hunk 15]
```diff
@@ -23,12 +24,15 @@ export const useCaseApp = {
 
   deleteUseCaseBy: async (field: UseCaseMutator): Promise<UseCase> => {
     const useCase = await useCaseDomain.loadUseCaseBy(field);
+    if (!useCase) {
+      throw new Error(NotFoundErrorCode.UseCaseNotFound);
```
Comment (Kroustille) [inline apps/backend/src/modules/use-case/use-case.app.ts]:
```suggestion
      throw new Error(ErrorCode.UseCaseNotFound);
```

File: apps/backend/src/modules/use-case/use-case.app.ts [hunk 16]
```diff
@@ -23,12 +24,15 @@ export const useCaseApp = {
...
+    }
     await objectUseCaseDomain.deleteObjectUseCaseBy({
       use_case_id: useCase.id,
     });
     const deletedUseCase = await useCaseDomain.deleteUseCase(field);
     if (!deletedUseCase) {
-      throw new Error(UnknownErrorCode.UnknownError);
+      throw new Error(NotFoundErrorCode.UseCaseNotFound);
```
Comment (Kroustille) [inline apps/backend/src/modules/use-case/use-case.app.ts]:
```suggestion
      throw new Error(ErrorCode.UseCaseNotFound);
```

File: apps/backend/src/modules/use-case/use-case.app.ts [hunk 17]
```diff
@@ -37,9 +41,9 @@ export const useCaseApp = {
     id: UseCaseId,
     input: EditUseCaseInput
   ): Promise<UseCase> => {
-    const updated = await useCaseDomain.updateUseCase(id, input);
+    const updated = await useCaseDomain.updateUseCase(id, stripNulls(input));
     if (!updated) {
-      throw new Error(UnknownErrorCode.UnknownError);
+      throw new Error(NotFoundErrorCode.UseCaseNotFound);
```
Comment (Kroustille) [inline apps/backend/src/modules/use-case/use-case.app.ts]:
```suggestion
      throw new Error(ErrorCode.UseCaseNotFound);
```

File: apps/backend/src/security/directive-graphql/validator/platform-token-validator.ts [hunk 18]
```diff
@@ -67,6 +68,9 @@ export const createPlatformTokenResolver = (originalResolve) => {
     const organization = await loadOrganizationFromPlatformIdAndTokenHeaders(
       portalContext.req
     );
+    if (!organization) {
+      throw NotFoundErrorCode.OrganizationNotFound;
```
Comment (Kroustille) [inline apps/backend/src/security/directive-graphql/validator/platform-token-validator.ts]:
```suggestion
      throw new Error(ErrorCode.OrganizationNotFound);
```

**Review by Kroustille — APPROVED** [review]

File: apps/backend/src/thirdparty/hubspot/hubspot.ts (hunk 5 above)
Comment (carinelebas) [reply apps/backend/src/thirdparty/hubspot/hubspot.ts:156]: No. TL;DR: let variables lose their narrowing inside closures; const variables preserve it.

Inside the .some() callback, TypeScript re-evaluates the type of deploymentRequest and sees it could still be FullyQualifiedDeploymentRequest | undefined — because the variable is declared with let and could theoretically be reassigned between the outer scope and the callback execution.
By assigning it to const request, TypeScript knows request can never be reassigned, so the narrowed non-undefined type is guaranteed to hold inside the .some() callback.
